src/box/memtx_space.c

```diff
@@ -1218,29 +1226,34 @@ memtx_space_build_index(struct space *src_space, struct index *new_index,
 		if (new_index->def->iid == 0)
 			tuple_ref(tuple);
 		/*
-		 * Remember the latest inserted tuple to
-		 * avoid processing yet to be added tuples
-		 * in on_replace triggers.
-		 */
-		state.cursor = tuple;
-		tuple_ref(state.cursor);
-		if (++count % MEMTX_DDL_YIELD_LOOPS == 0 &&
-		    memtx->state == MEMTX_OK)
-			fiber_sleep(0);
-		/*
-		 * Sleep after at least one tuple is inserted to test
-		 * on_replace triggers for index build.
-		 */
-		ERROR_INJECT_YIELD(ERRINJ_BUILD_INDEX_DELAY);
-		tuple_unref(state.cursor);
-		/*
-		 * The on_replace trigger may have failed
-		 * during the yield.
+		 * Let index be built in background if it is possible.
 		 */
-		if (state.rc != 0) {
-			rc = -1;
-			diag_move(&state.diag, diag_get());
-			break;
+		if (can_yield) {
```

Comment:
To reduce code movement I'd just use reversed condition + continue operator:
if (! can_yield)
    continue;